libs/SalesforceSDKCore/SalesforceSDKCore/Classes/UserAccount/SFUserAccountManager.h

```diff
@@ -513,6 +513,10 @@ Use this method to stop/clear any authentication which is has already been start
  */
 - (BOOL)handleIDPAuthenticationResponse:(NSURL *)url options:(nonnull NSDictionary *)options NS_SWIFT_NAME(handleIdentityProviderResponse(from:with:));
 
+- (BOOL)handleIDPAuthenticationResponse:(NSURL *)url options:(nonnull NSDictionary *)options completion:(nullable void (^)(void))completion NS_SWIFT_NAME(handleIdentityProviderResponse(from:with:completion:));
+
+- (void)initiateSPAuthentication:(NSString *)spConsumerKey redirectURI:(NSString *)spRedirectURI scopes:(NSSet<NSString *> *)spScopes;
```

Comment:
Where does the IDP get the sp consumer key and redirect uri?

Reply:
I thought the IDP app would know that and pass it in?

Reply:
You are right :-)
On Android, SalesforceSDKManager has a setAllowedSPApps(List<SPConfig> allowedSPApps) method.
To kick off a IDP initiated login flow, you just pass the SP app name (part of the SPConfig).
During SP initiated login flow, IDP only services a request from a SP app if it is one of the allowed ones.
But I guess SP initiated login is out of scope.

Reply:
ahh, I thought SP initiated would use the existing two hop flow since iOS doesn't have the broadcast option, did I misunderstand? also want me to make the params into an SPConfig object instead to match more?

Reply:
I think you are right. (I will end up having to re-introduce the two hops for Android also since the code cannot be obtained without user interactions in all cases).
Would the SP's access group name be part of the SPConfig then?

Reply:
Having the group on SPConfig sounds good -- so the IDP would use the group from the config and the SP would still set the group on SalesforceSDKManager so it can write the "loggedIn" key?